### DISCUSSION TITLE
Migrate UWP apps to Windows App SDK with the .NET Upgrade Assistant

### DISCUSSION THREAD

**Discussion by bgursahani** [microsoft/WindowsAppSDK#2990] [Announcements]
As a next step in assisting you to migrate your Universal Windows Platform (UWP) apps to the Windows App SDK and WinUI 3, we've leveraged the .NET Upgrade Assistant, adding support for migrating C# UWP apps. The UWP support automates much of the migration process. An updated preview is now available with features such as
 
•	Support for more APIs in domains such as Windowing, WinUI, AppLifecycle etc.
•	Support for WinRT components
•	Support for apps with a non-standard project structure
•	Post-upgrade report as soon as you culminate your migration journey
•	Improved error messages within the tool
•	Bug fixes 
 
Note: For more info such as how to get the tool, what to expect and things to know before starting, please see [Migrate UWP apps to Windows App SDK with the .NET Upgrade Assistant](https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-uwp-framework). The [Upgrade Assistant GitHub repository](https://github.com/dotnet/upgrade-assistant) documents troubleshooting tips and known issues. If you find any issues while using the tool, please report them in that same GitHub repository, tagging them with an area tag of 'UWP'.

**Comment by myokeeh** [comment]
Will this work on Background Tasks now? I originally used Template Studio to create my UWP apps and they had OOP background tasks. I have not been successful in getting those converted.

**Reply by myokeeh** [reply]
https://github.com/microsoft/CsWinRT/discussions/1243

**Reply by myokeeh** [reply]
Any updates on Background Tasks?
